src/main/java/me/karakelley/http/controllers/StaticFilesController.java

```diff
@@ -4,27 +4,46 @@
 import me.karakelley.http.Request;
 import me.karakelley.http.Response;
 import me.karakelley.http.Status;
+import me.karakelley.http.utility.HtmlGenerator;
 
+import java.io.File;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.List;
 
 public class StaticFilesController implements Controller {
+
   private final ArrayList availableActions = new ArrayList<>(Collections.singleton("GET"));
 
   public Response respond(Request request) {
+    String requestedResource = request.getPath();
+
     Response response = new Response();
     if (request.validRequestLine() && availableActions.contains(request.getMethod())) {
-      return displayFilesResponse(response);
+      return serveDirectoryResponse(response, requestedResource);
     } else {
       return new InvalidRequestController().respond(request);
     }
   }
 
-  private Response displayFilesResponse(Response response) {
+  private Response serveDirectoryResponse(Response response, String requestedResource) {
     response.setStatus(Status.OK);
-    response.setBody(String.join("\n", PublicDirectory.getFiles()));
-    response.setHeaders("Content-Type", "text/plain");
+    response.setBody(String.join("", getFilesAndDirectoryLinks(requestedResource)));
+    response.setHeaders("Content-Type", "text/html");
     response.setHeaders("Content-Length", String.valueOf(response.getBody().getBytes().length));
     return response;
   }
+
+  private ArrayList<String> getFilesAndDirectoryLinks(String resource) {
+    List<File> files = PublicDirectory.getDirectoriesAndFiles(resource);
+    ArrayList<String> directories = new ArrayList<>();
```

Comment:
nit: this variable might be more accurately named `resources`, since you could have files or directories